### ISSUE TITLE
TypedMoneyExtensions AmountToDecimal Fraction Digit handling

### ISSUE THREAD

**Issue by csi-lund** [commercetools/commercetools-dotnet-core-sdk-v2#177]
**Describe the bug**

When we migrated out codebase form SDK 1 to 2 we saw several failed tests when using the extension AmountToDecimal. 

In SDK 1 the AmountToDecimal factors in  FractionDigits when converting to decimal. [here](https://github.com/commercetools/commercetools-dotnet-core-sdk/blob/33f2e372cc673325a5719fd0cef8c12dd11a57f7/commercetools.Sdk/commercetools.Sdk.Domain/Common/Money.cs#L13)

In SDK 2 if its not high precision then the conversion assumed that its 2 fraction digits and divides by 100.  [here](https://github.com/commercetools/commercetools-dotnet-core-sdk-v2/blob/master/commercetools.Sdk/commercetools.Sdk.Api/Extensions/TypedMoneyExtensions.cs#L139)

This caused errors where we using the same calculation logic code on multiple currencies and precisions - particularly but not exclusive to korean which has 0 fraction digits.

```
       //SDK 1
        public override decimal AmountToDecimal()
        {
            return CentAmount / (decimal) Math.Pow(10, FractionDigits ?? 2);
        }


        // SDK 2 
        public static decimal AmountToDecimal(this ITypedMoney typedMoney)
        {
            IHighPrecisionMoney highPrecisionMoney = typedMoney as IHighPrecisionMoney;
            if (highPrecisionMoney != null)
            {
                return (decimal)highPrecisionMoney.PreciseAmount / (decimal)Math.Pow(10.0, highPrecisionMoney.FractionDigits);
            }

            return (decimal)typedMoney.CentAmount / 100m;
        }
```

**To Reproduce**

```
   [Test]
        public void CT_TypedMoney_ZeroFractionCulture_Success()
        {
            var money = new TypedMoney() { CentAmount = 1000, CurrencyCode = "ko-KR", FractionDigits = 0 };

            var result1 = commercetools.Sdk.Api.Models.Common.TypedMoneyExtensions.AmountToDecimal(money);
            var result2 = AmountToDecimalSDK1(money.CentAmount, money.FractionDigits);
            var result3 = AmountToDecimalModified(money);

            Assert.AreNotEqual(1000, result1, "DecimalAmount");
            Assert.AreEqual(1000, result2, "DecimalAmount");
            Assert.AreEqual(1000, result3, "DecimalAmount");
        }


        //SDK 1 adapted 

        public decimal AmountToDecimalSDK1(long centAmount, int? fractionDigits)
        {
            return centAmount / (decimal)Math.Pow(10, fractionDigits ?? 2);
        }

        //Modified
        public decimal AmountToDecimalModified(this ITypedMoney typedMoney)
        {
            IHighPrecisionMoney highPrecisionMoney = typedMoney as IHighPrecisionMoney;
            if (highPrecisionMoney != null)
            {
                return (decimal)highPrecisionMoney.PreciseAmount / (decimal)Math.Pow(10.0, highPrecisionMoney.FractionDigits);
            }

            return typedMoney.CentAmount / (decimal)Math.Pow(10, typedMoney.FractionDigits);
        }
```

**Expected behavior**
AmountToDecimal handles fraction digits

 - dotnet: NET6  & Standard 2.1
 - SDK:  commercetools.Sdk.Api" Version="4.1.0"

**Comment by jenschude** [comment]
I just released version 4.2.0 with a fix

Thanks for the report

**Comment by csi-lund** [comment]
Thanks for quick turnaround
